WordPress/openverse
# Allow runtime configuration of nginx listening port

## Fixes



Related to https://github.com/WordPress/openverse-infrastructure/pull/648

## Description


ECS FARGATE requires that container and host ports match on a container definition. In the related PR we tried to map Nuxt to a different host port, in order to allow Nginx to listen on 8443 so that the target group port doesn't need to change. We did that by juggling the _host_ ports, but we can't have different host and container ports. Nuxt lets you override the port by passing `-p`, but we need to enable runtime configuration of the Nginx port by using an environment variable.

The net change in this PR is that you should be able to pass `OPENVERSE_NGINX_LISTEN_PORT` environment variable to the container and have Nginx listen on a port other than 8080. Proof that it works is in the fact that it still listens to 8080 now, as that is set as the default for that variable in the container build.

## Testing Instructions

Run `just build frontend_nginx && just frontend/up` and confirm you can still access the `frontend_nginx` service at `localhost:50290`. You also need to run the frontend locally on port `8443` using `just frontend/dev`

## Checklist


- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
